src/wp-includes/class-wp-autoload.php

```diff
@@ -0,0 +1,516 @@
+<?php
+/**
+ * Autoloader for WordPress classes.
+ *
+ * Include this file if you'd like to avoid having to create your own autoloader.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Autoloader.
+ */
+final class WP_Autoload {
+
+	/**
+	 * An array of classes and their respective paths.
+	 *
+	 * Note: The classnames should be lowercased.
+	 *
+	 * @access private
+	 *
+	 * @var array
+	 */
+	const CLASSES_PATHS = array(
```

Comment:
By bundling this array inside the class definition does this not mean that regenerating it or loading it elsewhere for analysis becomes harder? A dedicated PHP file that returns the array would make all of that much easier, and if it need regenerating the entire file can be discarded and recreated without having to mess around with search replacements or PHP parsing of `class-wp-autoload.php`

Reply:
TBH I didn't even think about automating the classmap generation... It's not every day that we introduce new classes in Core, so my thinking was that it can be manually maintained without any additional maintenance overhead. Instead of adding `include_once`/`require_once` etc like we currently do, we'd just add it in the classmap.

But yes, we could definitely move the array to a separate file and then just include it.
One of the reasons a `const` was used is also performance as noted in [this comment](https://github.com/WordPress/wordpress-develop/pull/3470#discussion_r1119444603) by @schlessera.
I wonder if using a file would negate some of the performance improvements of this PR, since it would also mean we'd have to read an additional file :thinking:

Reply:
> By bundling this array inside the class definition does this not mean that regenerating it or loading it elsewhere for analysis becomes harder?

A unit test to ensure new classes are added to this array could be handy (I'm not sure what that would look like, sorry). Old habits die hard so it would help alert contributors using a require rather than the auto loader.

Reply:
Thank you @peterwilsoncc, test added 👍